cmd/explaintest/r/topn_push_down.result

```diff
@@ -218,3 +218,56 @@ Apply_17	9990.00	root	semi join, inner:Selection_21, equal:[eq(test.t1.a, x.a)]
         └─Limit_28	1.00	cop	offset:0, count:1
           └─Selection_27	1.00	cop	gt(test.t2.b, test.t1.b)
             └─TableScan_26	1.25	cop	table:t2, range:[-inf,+inf], keep order:false, stats:pseudo
+drop table if exists t;
+create table t(a int not null, index idx(a));
+explain select /*+ TIDB_INLJ(t2) */ * from t t1 join t t2 on t1.a = t2.a limit 5;
+id	count	task	operator info
+Limit_11	5.00	root	offset:0, count:5
+└─IndexJoin_15	5.00	root	inner join, inner:IndexReader_14, outer key:t1.a, inner key:t2.a
+  ├─TableReader_17	4.00	root	data:TableScan_16
+  │ └─TableScan_16	4.00	cop	table:t1, range:[-inf,+inf], keep order:false, stats:pseudo
+  └─IndexReader_14	10.00	root	index:IndexScan_13
+    └─IndexScan_13	10.00	cop	table:t2, index:a, range: decided by [t1.a], keep order:false, stats:pseudo
+explain select /*+ TIDB_INLJ(t2) */ * from t t1 left join t t2 on t1.a = t2.a where t2.a is null limit 5;
+id	count	task	operator info
+Limit_12	5.00	root	offset:0, count:5
+└─Selection_13	5.00	root	isnull(t2.a)
+  └─IndexJoin_17	5.00	root	left outer join, inner:IndexReader_16, outer key:t1.a, inner key:t2.a
+    ├─TableReader_19	5.00	root	data:TableScan_18
+    │ └─TableScan_18	5.00	cop	table:t1, range:[-inf,+inf], keep order:false, stats:pseudo
```

Comment:
It may look confusing `LeftOuterJoin` requires more outer rows than `InnerJoin` above, it is because we use estimation for InnerJoin while directly use join's expected count for LeftOuterJoin. How do you like the behavior here? @zz-jason @lamxTyler

Reply:
Maybe the outer join should also use estimation? `ExpectCnt` is just a maxiumn count.